src/gtk/webview_webkit2_extension.cpp

```diff
@@ -0,0 +1,426 @@
+                                 gpointer user_data)
+{
+    if (g_strcmp0(interface_name, WXGTK_WEB_EXTENSION_INTERFACE) != 0)
+    {
+        return;
+    }
+
+    wxWebViewWebKitExtension *extension = (wxWebViewWebKitExtension*)user_data;
+
+    if (g_strcmp0(method_name, "ClearSelection") == 0)
+    {
+        extension->ClearSelection(parameters, invocation);
+    }
+    else if (g_strcmp0(method_name, "DeleteSelection") == 0)
+    {
+        extension->DeleteSelection(parameters, invocation);
+    }
+    else if (g_strcmp0(method_name, "GetPageSource") == 0)
+    {
+        extension->GetPageSource(parameters, invocation);
+    }
+    else if (g_strcmp0(method_name, "GetPageText") == 0)
+    {
+        extension->GetPageText(parameters, invocation);
+    }
+    else if (g_strcmp0(method_name, "GetSelectedSource") == 0)
+    {
+        extension->GetSelectedSource(parameters, invocation);
+    }
+    else if (g_strcmp0(method_name, "GetSelectedText") == 0)
+    {
+        extension->GetSelectedText(parameters, invocation);
+    }
+    else if (g_strcmp0(method_name, "HasSelection") == 0)
+    {
+        extension->HasSelection(parameters, invocation);
+    }
+}
+
+static const GDBusInterfaceVTable interface_vtable = {
+    wxgtk_webview_handle_method_call,
+    NULL,
+    NULL
+};
+
+gboolean
+wxgtk_webview_dbus_peer_is_authorized(GCredentials *peer_credentials)
+{
+    static GCredentials *own_credentials = NULL;
+    GError *error = NULL;
+
+    if (!own_credentials)
+    {
+        own_credentials = g_credentials_new();
+    }
+
+    if (peer_credentials && g_credentials_is_same_user(peer_credentials, own_credentials, &error))
+    {
+        return TRUE;
+    }
+
+    if (error)
+    {
+        g_warning("Failed to authorize web extension connection: %s", error->message);
+        g_error_free(error);
+    }
+    return FALSE;
+}
+
+static gboolean
+wxgtk_webview_authorize_authenticated_peer_cb(GDBusAuthObserver *observer,
+                                              GIOStream *stream,
+                                              GCredentials *credentials,
+                                              wxWebViewWebKitExtension *extension)
+{
+    return wxgtk_webview_dbus_peer_is_authorized(credentials);
+}
+
+static void
+wxgtk_webview_dbus_connection_created_cb(GObject *source_object,
+                                         GAsyncResult *result,
+                                         wxWebViewWebKitExtension *extension)
+{
+    static GDBusNodeInfo *introspection_data = NULL;
+    GDBusConnection *connection;
+    guint registration_id;
+    GError *error = NULL;
+
+    if (!introspection_data)
+    {
+        introspection_data = g_dbus_node_info_new_for_xml(introspection_xml, NULL);
+    }
+
+    connection = g_dbus_connection_new_for_address_finish(result, &error);
+    if (error)
+    {
+        g_warning("Failed to connect to UI process: %s", error->message);
+        g_error_free(error);
+        return;
```

Comment:
Not needed.  If error is non-NULL, then connection is NULL:
https://developer.gnome.org/gio/stable/GDBusConnection.html#g-dbus-connection-new-for-address-finish